src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -48,6 +49,8 @@
 }
 
 
+# pylint: disable=too-many-instance-attributes, unidiomatic-typecheck, too-many-branches
+# pylint: disable=logging-too-many-args, fixme
```

Comment:
Using `type` rather than `isinstance` was intentional. Replacing  `type(area.strategy) == FinitePowerPlant` check with `isinstance`  as `FinitePowerPlant` inherits `CommercialStrategy`, using `isinstance` in this case would lead to classifying `CommercialStrategy` as   ´FinitePowerPlant` too, same goes for other cases too in this module. Using type helps to catch the main class excluding the derived classes

```
class CommercialStrategy:
    pass
class FinitePowerPlant(CommercialStrategy):
    pass

ib = FinitePowerPlant()

isinstance(ib, FinitePowerPlant)
>>> True
isinstance(ib, CommercialStrategy)
>>> True

```

Reply:
You can resolve it though with nested `if isinstance()` : 
```
if isinstance(strategy, CommercialStrategy):
    if isinstance(strategy, FinitePowerPlant):
       ....
    if isinstance(strategy, MarketMakerStrategy):
       ....
``` 
I also think it is more readable, because it reflects the class hierarchy in the if blocks, contrary to `type` which flattens these and the first-time viewer of the code is unaware that these types are related. 
Also, in your example you did not try this:
```
cs = CommercialStrategy()
isinstance(cs, FinitePowerPlant)
```
If the above returns false (which to my knowledge it will return false), then it is possible to deal the situation with isinstance and without type.

Reply:
We also discussed that whenever we come across TODO comments, we evaluate them and either remove them or fix them or add refactoring tickets. For this case there already is a refactoring ticket https://gridsingularity.atlassian.net/browse/D3ASIM-2288 which has been deprecated already (because we do not save unix times in the results but datetime strings anyway). Therefore the ticket should be closed, the TODO comment should be removed, and the pylint: disable=fixme should also be removed.